File: infrastructures/ddd/typescript/src/api/dto/v1/responses/ExampleResponse.ts
import { BaseResponseDto } from '../infra/BaseResponseDto';

/**
 * Response DTO for example data.
 * Response DTOs:
 * - Live in /api/dto/v1/responses/
 * - Suffix with "Response"
 * - Used for HTTP response payloads
 * - Transport only - NO business logic
 * 
 * This DTO is mapped from:
 * - ExampleReadEntity (for query operations)
 * - ExampleModel (for command operations)
 */
export class ExampleResponse extends BaseResponseDto {
  /**
   * The example ID
   */
  public Id: string;

  /**
   * The example name
   */
  public Name: string;

  /**
   * The example description
   */
  public Description: string;

  /**
   * The owner user ID
   */
  public OwnerId: string;

  /**
   * The owner name (from read entity only)
   */
  public OwnerName: string;

  /**
   * Active status
   */
  public IsActive: boolean;

  /**
   * Display name (computed)
   */
  public DisplayName: string;

  /**
   * Status text (computed)
   */
  public StatusText: string;

  /**
   * Creation timestamp
   */
  public CreatedAt: Date;

  /**
   * Last update timestamp
   */
  public UpdatedAt: Date;

  constructor() {
    super();
    this.Id = '';
    this.Name = '';
    this.Description = '';
    this.OwnerId = '';
    this.OwnerName = '';
    this.IsActive = true;
    this.DisplayName = '';
    this.StatusText = '';
    this.CreatedAt = new Date();
    this.UpdatedAt = new Date();
  }
}

File: infrastructures/ddd/csharp/src/Api/Dto/V1/ExampleResponse.cs
using System;

namespace OddlyDdd.Api.Dto.V1
{
    /// <summary>
    /// Example response DTO for returning example data.
    /// Response DTOs:
    /// - Live in /Api/Dto/
    /// - Have Response suffix
    /// - Used for HTTP transport only
    /// - NO business logic
    /// - Mapped from ReadEntity (queries) or BMO (commands)
    /// </summary>
    public class ExampleResponse : BaseResponseDto
    {
        public string Id { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string OwnerId { get; set; } = string.Empty;
        public string OwnerName { get; set; } = string.Empty;
        public bool IsActive { get; set; }
        public string DisplayName { get; set; } = string.Empty;
        public string StatusText { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
    }
}
